### PR TITLE
feat(amplify-util-uibuilder): support StorageField

### PR DESCRIPTION
**Pull request by hein-j**
#### Description of changes
- Update `amplify-util-uibuilder`'s dep on `aws-sdk` to 2.1350.0 ([changelog](https://github.com/aws/aws-sdk-js/blob/master/CHANGELOG.md#213500)) and `codegen-ui-react` and `codegen-ui` to `2.12.1` ([changelog for 2.12.0](https://github.com/aws-amplify/amplify-codegen-ui/releases/tag/v2.12.0), [changelog for 2.12.1](https://github.com/aws-amplify/amplify-codegen-ui/releases/tag/v2.12.1)). Both updates are needed to support `StorageField` in form builder.
- In updating`aws-sdk`, update all mentions of the dependency in the monorepo to make sure it all resolves to the same version.
- Update dependency warning tests to reflect the new recommendation to use `@aws-amplify/ui-react-storage`

#### Description of how you validated changes
- Created a form with `StorageField`, pulled it down, and checked the generated jsx.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-util-uibuilder/package.json [hunk 1]
```diff
@@ -16,10 +16,10 @@
   "dependencies": {
     "@aws-amplify/amplify-cli-core": "4.0.1",
     "@aws-amplify/amplify-prompts": "2.6.6",
-    "@aws-amplify/codegen-ui": "2.11.1",
-    "@aws-amplify/codegen-ui-react": "2.11.1",
+    "@aws-amplify/codegen-ui": "2.12.0",
+    "@aws-amplify/codegen-ui-react": "2.12.0",
     "amplify-codegen": "^3.4.1",
-    "aws-sdk": "^2.1233.0",
+    "aws-sdk": "^2.1350.0",
```
Comment (sobolk) [inline packages/amplify-util-uibuilder/package.json:22]: why do we need to bump aws-sdk?
if we want to do this we should make sure all packages in cli resolve same version.

Comment (hein-j) [reply packages/amplify-util-uibuilder/package.json:22]: > why do we need to bump aws-sdk?
We need to bump the version to take advantage of the `StorageField` configurations we just added to the SDK. 
> if we want to do this we should make sure all packages in cli resolve same version.
I am seeing that the last time our team updated the SDK version, we did this by updating all the versions in the monorepo. I will rev with that change now.

File: packages/amplify-nodejs-function-template-provider/resources/lambda/crud/package.json.ejs [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+  "body-parser": "^1.19.1",
+  "express": "^4.17.2"
   },
   "devDependencies": {
-    "@types/aws-lambda": "^8.10.92",
-    "aws-sdk": "^2.1113.0"
+  "@types/aws-lambda": "^8.10.92",
+  "aws-sdk": "^2.1350.0",
+  }
   }
```
Comment (sobolk) [inline packages/amplify-nodejs-function-template-provider/resources/lambda/crud/package.json.ejs:16]: please revert changes in this file. this is a template for lambda functions, not related to what CLI is using.

**Review by sobolk — COMMENTED** [review]
We also have dependency on sdk in category api which is not here.
Can you add
`"**/aws-sdk": "^2.1350.0"`
to this section https://github.com/aws-amplify/amplify-cli/blob/9eab404a232a4ecc5cb14c8b063e7d217bea3492/package.json#L152-L162 ?

**Comment by sobolk** [comment]
Looks good now. Can you run e2e tests?

File: packages/amplify-nodejs-function-template-provider/resources/lambda/crud/package.json.ejs (hunk 2 above)
Comment (hein-j) [reply packages/amplify-nodejs-function-template-provider/resources/lambda/crud/package.json.ejs]: Reverted the change.

**Comment by hein-j** [comment]
> We also have dependency on sdk in category api which is not here. Can you add `"**/aws-sdk": "^2.1350.0"` to this section
> 
> https://github.com/aws-amplify/amplify-cli/blob/9eab404a232a4ecc5cb14c8b063e7d217bea3492/package.json#L152-L162
> 
> ?

Added the dependency.

**Comment by hein-j** [comment]
> 

Yes, thanks a lot for the review.

**Review by sobolk — APPROVED** [review]

**Review by sdstolworthy — APPROVED** [review]

**Pull request merged**